src/users/user.php

```diff
@@ -319,19 +328,21 @@ public function delete()
 	/**
 	 * Function retriveUser for get user details depending upon the identifier.
 	 *
-	 * @param   string  $xidentifier     Flag to differentiate the column value.
-	 *
 	 * @param   string  $userIdentifier  username
 	 *
 	 * @return  object  $user  Juser object if user exist otherwise std class.
 	 *
 	 * @since   2.0
 	 */
-	private function retriveUser($xidentifier, $userIdentifier)
+	private function retriveUser($userIdentifier)
 	{
 		$user = new stdClass;
 
-		switch ($xidentifier)
+		// Flag to differentiate the column value
+		$app            = JFactory::getApplication();
+		$xIdentifier    = $app->input->server->get('HTTP_X_IDENTIFIER', '', 'string');
```

Comment:
Does the type need to be string ? The values are only `email` or `username` right ?

Reply:
yes.. id, username and email.. changing it to WORD

Reply:
The default is CMD which is sufficient

Reply:
CMD  Allow a-z, 0-9, underscore, dot, dash. Also remove leading dots from result.